fortran95.html.markdown

```diff
@@ -0,0 +1,445 @@
+versions are FORTRAN 77, Fortran 90, Fortran 95, Fortran 2003, Fortran 2008 and
+Fortran2015.
+
+This overview will discuss the features of Fortran 95 since it is the most
+widely implemented of the more recent specifications and the later versions are
+largely similar (by comparison FORTRAN 77 is a very different language).
+
+```fortran
+
+! This is a comment.
+
+! Code can only exist inside programs, functions, subroutines or modules.
+
+! long lines can be continued using the continuation character '&'
+
+program example   !declare a program called example.
+
+    ! Indentation is not actually required but is recommended.
+
+    ! Declaring variables
+    ! ===================
+    
+    ! All declarations must come before statements and expressions.
+    
+    implicit none    !prevents dynamic declaration of variables (recommended!)
+    ! Implicit none must be redeclared in every function/program/module...
+    
+    ! IMPORTANT - Fortran is case insensitive.
+    real z
+    REAL Z2
+
+    real :: v,x    ! WARNING: default initial values are compiler dependent!    
+    real :: a = 3, b=2E12, c = 0.01
+    integer :: i,j,k=1,m
+    real, parameter :: PI = 3.1415926535897931    !declare a constant.
+    logical :: y = .TRUE. , n = .FALSE.    !boolean type.
+    complex :: w = (0,1)    !sqrt(-1)
+    character (len=3) :: month    !string of 3 characters.
+    
+    real :: array(6)     !declare an array of 6 reals.
+    real, dimension(4) :: arrayb    !another way to declare an array.
+    integer :: arrayc(-10:10)   !an array with a custom index.
+    real :: array2d(3,2)    !multidimensional array.
+    
+    ! The '::' separators are not always necessary but are recommended.
+
+    ! many other variable attributes also exist:
+    real, pointer :: p    !declare a pointer.
+
+    integer, parameter :: LP = selected_real_kind(20)
+    real (kind = LP) :: d    !long precision variable.
+
+    ! WARNING: initialising variables during declaration causes problems
+    ! in functions since this automatically implies the 'save' attribute
+    ! whereby values are saved between function calls. In general, separate
+    ! declaration and initialisation code except for constants!
+    
+    
+    ! Strings
+    ! =======
+
+    character :: a_char = 'i'
+    character (len = 6) :: a_str = "qwerty"
+    character (len = 30) :: str_b
+    character (len = *), parameter :: a_long_str = "This is a long string."
+    !can have automatic counting of length using (len=*) but only for constants.
+    
+    str_b = a_str // " keyboard"    !concatenate strings using // operator.
+
+
+    ! Assignment & Arithmetic
+    ! =======================
+
+    Z = 1    !assign to variable z declared above (case insensitive).
+    j = 10 + 2 - 3
+    a = 11.54  /  (2.3 * 3.1)
+    b = 2**3    !exponentiation
+    
+    !an aside:
+    print *, b    !print the variable 'b' to the command line (see I/O section).
```

Comment:
I wanted to introduce it early without going into (in my opinion) the distracting details of the print statement, too early in the tutorial. But I agree that it isn't very logical to put it here.